mkosi/config.py

```diff
@@ -2778,7 +2778,10 @@ def json_transformer(key: str, val: Any) -> Any:
         # It is unlikely that the type of a field will be None only, so let's not bother with a different sentinel
         # value
         if fieldtype is None:
-            ValueError(f"{refcls} has no field {key}")
+            try:
+                ValueError(f"{refcls} has no field {key}")
+            except (ValueError, IndexError, AssertionError) as e:
+                raise ValueError(f"Unable to parse {val:r} for attribute {key:r} for {refcls.__name__}") from e
```

Comment:
Not sure I understand what's going on here?

Reply:
Copy paste error from above. The idea is to generate a single known error when a transformer fails..